nwbinspector/inspector_tools.py

```diff
@@ -1,34 +1,80 @@
 """Internally used tools specifically for rendering more human-readable output from collected check results."""
-import sys
 import os
-from enum import Enum
-from collections import OrderedDict
+import sys
 from typing import Dict, List
 from pathlib import Path
+from natsort import natsorted
+from enum import Enum
 
-import numpy as np
-
-from .register_checks import Importance
+from .register_checks import InspectorMessage
 from .utils import FilePathType
 
 
-def sort_by_descending_severity(check_results: list):
-    """Order the dictionaries in the check_list by severity."""
-    severities = [check_result.severity.value for check_result in check_results]
-    descending_indices = np.argsort(severities)[::-1]
-    return [check_results[j] for j in descending_indices]
+def _sort_unique_values(unique_values: list):
+    """Technically, the 'set' method applies basic sorting to the unique contents, but natsort is more general."""
+    if any(unique_values) and isinstance(unique_values[0], Enum):
+        return natsorted(unique_values, key=lambda x: -x.value)
+    else:
+        return natsorted(unique_values)
+
 
+def organize_messages(messages: List[InspectorMessage], levels: List[str]):
+    """
+    General function for organizing list of InspectorMessages.
+
+    Returns a nested dictionary organized according to the order of the 'levels' argument.
+
+    Parameters
+    ----------
+    messages : list of InspectorMessages
+    levels: list of strings
+        Each string in this list must correspond onto an attribute of the InspectorMessage class, excluding the
+        'message' text.
+    """
```

Comment:
@bendichter Amongst your suggestions for this new function was a `reverse` option for flipping the direction of ordering along a particular nesting level.

(a) Can you think of situations where this might be wanted by a user or ourselves?

(b) Did you want that exposed to the CLI as well (TODO in a future PR)? If so what tag, `-r'/`--reverse`? Encoded as a comma-separated string equal in length to `--levels`?

Reply:
Yes this is needed, some we want the results to show up from high to low. Another option is to do the reversing in the report creation function

Reply:
I'll add reverse to the organization, then, but not to the CLI